src/SolverAPI.jl

```
@@ -462,32 +462,74 @@ json_to_snf(a::Real, ::Dict) = a
 nl_to_aff_or_quad(::Type{T}, f::T) where {T<:Real} = f
 nl_to_aff_or_quad(::Type{T}, f::Real) where {T<:Real} = convert(T, f)
 
+function _construct_saf_or_qd(::Type{T}, op::Symbol, args_rev::Vector) where {T<:Real}
+    # Note that `args_rev` are in reverse order.
+    if op == :^
+        if length(args_rev) == 2 && args_rev[1] == 2
+            return MOI.Utilities.operate(*, T, args_rev[2], args_rev[2])
+        end
+    else
+        if op == :+
+            # NOTE (dba) this is a workaround to avoid a
+            # `StackOverflowError` error coming from
+            # `MOI.Utilities.operate` for large `args`. It is
+            # recursively defined:
+            # https://github.com/jump-dev/MathOptInterface.jl/blob/master/src/Utilities/operate.jl#L323-L327
+            #
+            # NOTE (dba) It's important we use the in-place version to
+            # reduce allocations!
+            #
+            # As `:+` is commutative we don't need to worry about the
+            # order of `args_rev`.
+            plus_op(accum, x) = MOI.Utilities.operate!(+, T, accum, x)
+            return reduce(plus_op, args_rev)
+        else
+            h = get(_quad_ops, op, nothing)
+            # All other operators do not take varargs. See
+            # https://github.com/jump-dev/MathOptInterface.jl/blob/master/src/Utilities/operate.jl#L329
+            if !isnothing(h)
+                # TODO (dba) convert this assertion into a validation.
+                @assert length(args_rev) == 2
+                return MOI.Utilities.operate(h, T, args_rev[2], args_rev[1])
+            end
+        end
+    end
+    throw(Error(Domain, "Function cannot be converted to linear or quadratic form."))
+end
+
 _quad_ops = Dict(:+ => +, :- => -, :* => *, :/ => /)
 
 function canonicalize_SNF(::Type{T}, f) where {T<:Real}
-    try
-        f = nl_to_aff_or_quad(T, f)
-    catch
-    end
-    return f
+    return nl_to_aff_or_quad(T, f)
```

This hides the underlying exception and makes it harder to debug. @chriscoey Are there cases where we can actually continue even though we can't convert the SNF?

---

Nevermind I found the reason :)